### PR TITLE
Develop lp

### PR DESCRIPTION
**Pull request by paulstapor**
If you have the time at some point, here is a pull-request from what we (Leonard and I) have changed on Memoir in our version so far.

The main interest of this pull request is not necessarily to merge it now, rather it is to see what has changed so far... and what should still be changed. Feel free to reject it.

Most probably, the changes in MEMfn/ could be omitted. I kept them here for simplicity.

### CODE REVIEW

File: getSimulationPA.m [hunk 1]
```diff
@@ -15,29 +19,70 @@
     else
         op_SP.approx = 'sp';
     end
-%     SP = testSigmaPointApp(@(phi) simulateForSP(Model.exp{s}.model,Data{s}.PA.time,phi,Data{s}.condition),xi,Model.exp{s},op_SP);
```
Comment (FFroehlich) [inline getSimulationPA.m:20]: ugh, some things from develop branch are included here and some are not :/ not really ideal
please set scale to lin op_SP.approx is not 'sp'

File: objective_phi.m [hunk 2]
```diff
@@ -2,7 +2,7 @@
 
 Ym = data.SCTL.Y(:,:,i);
 ind_y = data.SCTL.ind_y(:,i);
-ny = sum(ind_y);
+ny = sum(ind_y)*size(Ym,2);
```
Comment (FFroehlich) [inline objective_phi.m]: are the problem that occur here fixed in #2 ?

**Review by FFroehlich — CHANGES_REQUESTED** [review]

**Comment by FFroehlich** [comment]
please use more descriptive commit messages "little changes" and "minor changes" are almost equivalent to no commit message at all.

**Comment by FFroehlich** [comment]
this pull request is continually growing and conflicts are piling up. please get to a functional version and merge as soon as possible. smaller chunks are easier to review and less error-prone for merging.

Also please adress the comments i had on the code (or at least comment that this is done/why it is not necessary).

File: complete_model.m [hunk 3]
```diff
@@ -171,7 +174,19 @@
...
+            end
+            l = 1;
+            for j = 1:n_b
+                for k = 1:j-1
+                    C(k,j) = Model.sym.delta(n_b + l);
+                    C(j,k) = Model.sym.delta(n_b + l);
+                    l = l + 1;
+                end
+            end
+        case {'matrix-logarithm', 'cholesky-parametrization'}
```
Comment (FFroehlich) [inline complete_model.m:189]: no filename for cholesky?

File: complete_model.m [hunk 4]
```diff
@@ -260,7 +280,7 @@
         Model.exp{s}.sym.dsigma_noisedphi = sym(zeros(size(Model.exp{s}.sym.sigma_noise,1),size(Model.exp{s}.sym.sigma_noise,2),n_phi));
         for j = 1:size(Model.exp{s}.sym.sigma_noise,1)
             for k = 1:size(Model.exp{s}.sym.sigma_noise,2)
-                Model.exp{s}.sym.dsigma_noisedphi(j,k,:) = jacobian(Model.exp{s}.sym.sigma_noise(j,k),phi);
+                % Model.exp{s}.sym.dsigma_noisedphi(j,k,:) = jacobian(Model.exp{s}.sym.sigma_noise(j,k),phi);
```
Comment (FFroehlich) [inline complete_model.m]: remove comment?

File: make_model.m [hunk 5]
```diff
@@ -54,7 +54,7 @@
 end
 
 switch(Model.type_D)
-    case 'matrix-logarithm'
+    case {'matrix-logarithm', 'givens-parametrization', 'Lie-generators'}
```
Comment (FFroehlich) [inline make_model.m:57]: cholesky?

**Review by FFroehlich — APPROVED** [review]

File: make_model.m (hunk 5 above)
Comment (paulstapor) [reply make_model.m:57]: Not yet implemented, but it's on the list for a little study on efficiency. However, first things first...

**Review by DantongWang — APPROVED** [review]

**Pull request merged**